aiohttp_client_cache/session.py

```diff
@@ -50,40 +67,52 @@ async def _request(
     ) -> CachedResponse:
         """Wrapper around :py:meth:`.SessionClient._request` that adds caching"""
         # Attempt to fetch cached response
-        response, actions = await self.cache.request(
-            method, str_or_url, expire_after=expire_after, refresh=refresh, **kwargs
+        key = self.cache.create_key(method, str_or_url, **kwargs)
+        actions = self.cache.create_cache_actions(
+            key, str_or_url, expire_after=expire_after, refresh=refresh, **kwargs
         )
 
-        def restore_cookies(r):
-            self.cookie_jar.update_cookies(r.cookies or {}, r.url)
-            for redirect in r.history:
-                self.cookie_jar.update_cookies(redirect.cookies or {}, redirect.url)
-
-        if actions.revalidate and response:
-            from_cache, new_response = await self._refresh_cached_response(
-                method, str_or_url, response, actions, **kwargs
-            )
-            if not from_cache:
-                return set_response_defaults(new_response)
-            else:
-                restore_cookies(new_response)
-                return cast(CachedResponse, new_response)
-
-        # Restore any cached cookies to the session
-        if response:
-            restore_cookies(response)
-            return response
-        # If the response was missing or expired, send and cache a new request
-        else:
-            if actions.skip_read:
-                logger.debug(f'Reading from cache was skipped; making request to {str_or_url}')
+        lock: Lock | nullcontext = self._null_lock
+        if not actions.skip_read:
+            try:
+                lock = self._locks[key]
+            except KeyError:
+                self._locks[key] = Lock()
+                lock = self._locks[key]
```

Comment:
I think this section+ `_null_lock` could be replaced with [defaultdict](https://docs.python.org/3/library/collections.html), for example:
```py
self._locks = defaultdict(Lock)
```

That creates a new `Lock` object for `self._locks[key]` it doesn't already exist.

Reply:
Thank you, the more I know. 
But for null context, I use that because if the cache is disabled we don't need to lock the request, because if we lock it, the concurrent request will execute sequentially.